indispensable/src/commonMain/kotlin/at/asitplus/signum/indispensable/asn1/Asn1Elements.kt

```diff
@@ -127,6 +127,62 @@ sealed class Asn1Element(
         return result
     }
 
+    /**
+     * Convenience function to cast this element to an [Asn1Primitive]
+     * @throws ClassCastException if this element is not a primitive
+     */
+    @Throws(ClassCastException::class)
+    fun asPrimitive(): Asn1Primitive = casting { this as Asn1Primitive }
```

Comment:
I'm not sure i see the use case for these functions; how is `foo.asPrimitive()` different from `(foo as Asn1Primitive)`?

Reply:
you won't require and braces and spaces, making chaining smoother

Reply:
should this maybe be `assertPrimitive()` or similar? to me, "asX" doesn't feel like a function that might throw if it's not an X... (compare `asList`, `asSet`, etc.)

Reply:
it says so in the docs and it is annotated and it is the same pattern used by kotlinx.serialization, so I'd expect our target audience to be familiar with it.

Reply:
you know i'm not fond of "it says so in the docs"

Reply:
i'd still be more comfortable with `.assertX` (and this would even show up in IDE autocompletion for `asX`)